### PR TITLE
[CSP] test that quote characters other than U+0027 APOSTROPHE don't work

### PR DESCRIPTION
**Pull request by foolip**
Part of https://github.com/w3c/webappsec-csp/issues/434.

### CODE REVIEW

File: content-security-policy/generic/quote-characters.html [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+<head>
+  <script src="/resources/testharness.js"></script>
+  <script src="/resources/testharnessreport.js"></script>
+</head>
+<body>
+  <script>
+    var tests = [
+      { "csp": "img-src 'none';", "expected": false, "name": "precondition: single quotes are accepted" },
+      // See https://github.com/w3c/webappsec-csp/issues/434 for background on what is tested.
+      { "csp": 'img-src "none";', "expected": true, "name": "double quotes should not be accepted" },
```
Comment (mikewest) [inline content-security-policy/generic/quote-characters.html:12]: In Chromium, this treats the source `"none"` as an unknown source, leaving us with `img-src`, which is an empty list, which is the same as `'none'`.

File: content-security-policy/generic/quote-characters.html [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      { "csp": "img-src \u0091none'", "expected": true, "name": "U+0091 + single quote should not be accepted" },
+      { "csp": "img-src 'none\u0092", "expected": true, "name": "single quote + U+0092 should not be accepted" },
+      { "csp": "img-src \u0091none\u0092", "expected": true, "name": "U+0091 + U+0092 should not be accepted" },
+      { "csp": "img-src \u2018none\u2019;", "expected": true, "name": "U+2018 + U+2019 should not be accepted" },
```
Comment (mikewest) [inline content-security-policy/generic/quote-characters.html:16]: In Chromium, these four all end up as parse failures for the directive, as it contains characters that can't exist in a directive. In this case, we punt the directive entirely. We should probably have the same behavior here that we have for `"none"` above: treat the policy as containing an `img-src` directive with no valid entries, which is equivalent to `'none'`.

**Review by mikewest — COMMENTED** [review]
@ArthurSonzogni to weigh in on whether we can reasonably change Chromium's behavior.

**Comment by ArthurSonzogni** [comment]
I checked the sources. Chrome consider all those "sources" to be invalid. They are ignored. As a result the source list is empty and behave as "img-src 'none'".
A console error message is emitted to warn the developers.

I consider this behavior to be okay. I recognize that this is a bit confusing when the invalid source is "none".

What would be the new behavior? To drop the directive entirely? (we currently only drop the invalid source)

**Comment by foolip** [comment]
Abandoning before the 3 year anniversary of this PR, I'm realistically not going to come back to it.

**Pull request closed without merging**